delphi/LeapfrogInterface.pas

```diff
@@ -0,0 +1,955 @@
+    hc2ArtAidsDeathsLength: Integer;
+    hcArtInit: PDouble;
+    hcArtInitLength: Integer;
+    hcArtNeedInit: PDouble;
+    hcArtNeedInitLength: Integer;
+    hivBirths: PDouble;
+    hivBirthsLength: Integer;
+    ctxNeed: PDouble;
+    ctxNeedLength: Integer;
+    infectionByType: PDouble;
+    infectionByTypeLength: Integer;
+  public
+    procedure SetHc1HivPop(var inHc1HivPop: TGBFixedArray<Double>);
+    procedure SetHc2HivPop(var inHc2HivPop: TGBFixedArray<Double>);
+    procedure SetHc1ArtPop(var inHc1ArtPop: TGBFixedArray<Double>);
+    procedure SetHc2ArtPop(var inHc2ArtPop: TGBFixedArray<Double>);
+    procedure SetHc1NoartAidsDeaths(var inHc1NoartAidsDeaths: TGBFixedArray<Double>);
+    procedure SetHc2NoartAidsDeaths(var inHc2NoartAidsDeaths: TGBFixedArray<Double>);
+    procedure SetHc1ArtAidsDeaths(var inHc1ArtAidsDeaths: TGBFixedArray<Double>);
+    procedure SetHc2ArtAidsDeaths(var inHc2ArtAidsDeaths: TGBFixedArray<Double>);
+    procedure SetHcArtInit(var inHcArtInit: TGBFixedArray<Double>);
+    procedure SetHcArtNeedInit(var inHcArtNeedInit: TGBFixedArray<Double>);
+    procedure SetHivBirths(var inHivBirths: TGBFixedArray<Double>);
+    procedure SetCtxNeed(var inCtxNeed: TGBFixedArray<Double>);
+    procedure SetInfectionByType(var inInfectionByType: TGBFixedArray<Double>);
+end;
+
+{$ALIGN 8}
+type
+  LeapfrogParams = record
+  private
+    dp: ^LeapfrogDpParams;
+    ha: ^LeapfrogHaParams;
+    hc: ^LeapfrogHcParams;
+  public
+    procedure SetDpParams(var dpParams: LeapfrogDpParams);
+    procedure SetHaParams(var haParams: LeapfrogHaParams);
+    procedure SetHcParams(var hcParams: LeapfrogHcParams);
+end;
+
+{$ALIGN 8}
+type
+  LeapfrogState = record
+  private
+    dp: ^LeapfrogDpState;
+    ha: ^LeapfrogHaState;
+    hc: ^LeapfrogHcState;
+  public
+    procedure SetDpState(var dpState: LeapfrogDpState);
+    procedure SetHaState(var haState: LeapfrogHaState);
+    procedure SetHcState(var hcState: LeapfrogHcState);
+end;
+
+type TCallbackFunction = procedure(Msg: PAnsiChar); stdcall;
+
+procedure LeapfrogRunDp(var params: LeapfrogParams; var opts: LeapfrogOptions; var outputState: LeapfrogState; errorHandler: TCallbackFunction); safecall; external 'leapfrog.dll' name 'run_dp';
+procedure LeapfrogRunAim(var params: LeapfrogParams; var opts: LeapfrogOptions; var outputState: LeapfrogState; errorHandler: TCallbackFunction); safecall; external 'leapfrog.dll' name 'run_aim';
+procedure LeapfrogRunDpSingleYear(var params: LeapfrogParams; var opts: LeapfrogOptions; var initial_state: LeapfrogState; start_year: Integer; var state: LeapfrogState; errorHandler: TCallbackFunction); safecall; external 'leapfrog.dll' name 'run_dp_single_year';
+procedure LeapfrogRunAimSingleYear(var params: LeapfrogParams; var opts: LeapfrogOptions; var initial_state: LeapfrogState; start_year: Integer; var state: LeapfrogState; errorHandler: TCallbackFunction); safecall; external 'leapfrog.dll' name 'run_aim_single_year';
```

Comment:
We need a function to set up this `initial_state` for the first year. See what leapfrog full model does, this is separate from the base year calculation in spectrum desktop to back project in the first year.

Reply:
Added an issue for this #189 I am thinking that probably this will all get handled in spectrum desktop down the line once we work out how to run year-by-year. But let's discuss it.